src/test/java/EnergyTest.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.Test;
+
+import java.awt.*;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class EnergyTest {
+
+    @Test
+    void energy4x4() {
+        //given
+        Color color1 = new Color(255, 0, 0);
+        Color color2 = new Color(0, 255, 0);
+        Color color3 = new Color(0, 0, 255);
+
+        Pixel[][] image = new Pixel[][]{
+                {new Pixel(color1.getRGB()),
+                        new Pixel(color2.getRGB()),
+                        new Pixel(color3.getRGB()),
+                        new Pixel(color1.getRGB())},
+                {new Pixel(color2.getRGB()),
+                        new Pixel(color3.getRGB()),
+                        new Pixel(color1.getRGB()),
+                        new Pixel(color1.getRGB())},
+                {new Pixel(color3.getRGB()),
+                        new Pixel(color1.getRGB()),
+                        new Pixel(color1.getRGB()),
+                        new Pixel(color2.getRGB())},
+                {new Pixel(color1.getRGB()),
+                        new Pixel(color1.getRGB()),
+                        new Pixel(color2.getRGB()),
+                        new Pixel(color3.getRGB())}};
+        Energy energy = new Energy();
+        energy.updateCellEnergy(image);
+        double[][] calculatedEnergy = new double[image.length][image[0].length];
+        for (int row = 0; row < calculatedEnergy[0].length; row++)
+        {
+            for (int col = 0; col < calculatedEnergy.length; col++)
+            {
+                calculatedEnergy[row][col] = image[row][col].getCellEnergy();
+            }
+        }
+        //when
+        double[][] expected = new double[][]{
+                {255.0, 255.0, 255.0, 255.0},
+                {255.0, 0.0, 0.0, 255.0},
+                {255.0, 0.0, 0.0, 255.0},
+                {255.0, 255.0, 255.0, 255.0}};
+
+        //then
+        assertTrue(Arrays.deepEquals(expected, calculatedEnergy));
+    }
+
+
+    @Test
+    void calculateEnergy() {
+        //given
+        Color color1 = new Color(255, 0, 0);
+        Color color2 = new Color(0, 255, 0);
+        Color color3 = new Color(0, 0, 255);
```

Comment:
Also, there are already constants for these colors
Color.RED, Color.BLUE, Color.GREEN